pkg/controller/storagecluster/reconcile.go

```diff
@@ -744,11 +745,33 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string) *cephv1.CephClus
 		},
 	}
 
-	if len(sc.Spec.MonDataDirHostPath) != 0 {
+	// If both `MonDataDirHostPath` and `monPVCTemplate` is provided, the mons will
+	// be provisioned on the provided `monPVCTemplate` and `monDataDirHostPath`.
+	if len(sc.Spec.MonDataDirHostPath) != 0 && sc.Spec.MonPVCTemplate != nil {
```

Comment:
I don't think this case makes sense. In the case of both of these being configured we should probably just take the PVC template and not set the data dir hostpath. This would also be grounds for an admission controller to prevent this configuration, but that's in the future. I think the cases should just be:

1. If PVC template not nil (give it preference in the ordering, so data dir hostpath is ignored if this is set)
2. If data dir hostpath not empty
3. If StorageDeviceSets exist
4. Else, warning (as Rohan said), and set nothing, letting Rook-Ceph take its default behavior

Reply:
yup updated the PR

Reply:
@jarrpa why shouldn't we allow setting the dataDirHostPath even if mons are not using hostPath? Rook still ends up using the default `/var/lib/rook` for OSD.

 Seems like we're getting it for free with this new field.